chalk-rust-ir/src/lib.rs

```diff
@@ -99,6 +103,15 @@ pub struct TraitDatum<I: Interner> {
 
     /// The id of each associated type defined in the trait.
     pub associated_ty_ids: Vec<AssocTypeId<I>>,
+
+    /// A marker indicating if this trait definition represents one of the
+    /// various builtin traits (sized, copy, etc)
+    pub well_known: Option<WellKnownTrait>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
```

Comment:
A list of the traits that are "well known" to chalk, which means that the chalk-solve crate has special, hard-coded impls for them.